### ISSUE TITLE
``comparison-with-callable`` false positive then type of variable is a Callable

### ISSUE THREAD

**Issue by LefterisJP** [pylint-dev/pylint#3987]
Small note: You guys are awesome. Thanks a lot for pylint :hugs: 

### Steps to reproduce

```python
import operator
from typing import Callable


def foo(op: Callable = operator.add) -> None:
    if op == operator.sub:
        print('do something with sub')
    elif op == operator.add:
        print('do something with add')
    else:
        print('no idea')
```

### Current behavior

Getting:

> test2.py:6:7: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
> test2.py:8:9: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)

### Expected behavior

I would expect that this error should not be returned since it's obviously an intended comparison with a callable. You want to do something depending on the `callable` type that `op` is. So no need for parentheses as suggested by [the example](https://vald-phoenix.github.io/pylint-errors/plerr/errors/basic/W0143.html) here.

### pylint --version output

>pylint 2.6.0
> astroid 2.4.2
> Python 3.7.9 (default, Sep 26 2020, 15:11:06) 
> [GCC 10.2.0]

**Comment by PCManticore** [comment]
Thanks for the report, I can confirm this is a bug.

**Comment by jacobtylerwalls** [comment]
Can't reproduce on main (88543500c70e3aa303b6c0861e2a7a006673f6a8)
